File: c/2.c
// prints sum of all even-valued fibonacci numbers up to 4000000

#include <stdio.h>

int fibSum(int);

int main(int argc, char** argv) {
    int a = 1;
    int b = 2;
    int sum = 0;
    int temp;

    while (a < 4000000) {
        if (a % 2 == 0) {
            sum += a;
        }

        temp = a;
        a = b;
        b = temp + b;
    }

    printf("%d\n", sum);
    return 0;
}

File: js/2.js
// prints sum of all even-valued fibonacci numbers up to 4000000

var a = 1;
var b = 2;
var sum = 0;

while (a <= 4000000) {
    if (a % 2 == 0) {
        sum += a;
    }

    var temp = a;
    a = b;
    b = temp + b;
}

console.log(sum);
